packages/react-button/src/utils/useToggleState.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { useControllableState, useMergedEventCallbacks } from '@fluentui/react-utilities';
+import type { ButtonState } from '../Button';
+import type { ToggleButtonProps, ToggleButtonState } from '../ToggleButton';
+
+export function useToggleState<
+  TToggleButtonProps extends ToggleButtonProps,
+  TButtonState extends Partial<ButtonState>,
+  TToggleButtonState extends ToggleButtonState
```

Comment:
Agreed. Because of `Partial<ButtonState>` you have to explicitly handle a null case for `state.root` on L12 but the root slot is always `NonNullable` so this check should not be required. `Pick` should also eliminate the need for the type assertion on L44.

Reply:
I agree, I've changed this.